File: docs/legacy(old PACE)/unsupported/VB.NET/PaceWebServiceSDK/sample/CloseJob.vb
Imports PaceWebServiceSDK.efipaceservices
Imports System.Web.Services.Protocols

Public Class CloseJob

    Public Shared Sub Run()

        'read inputs
        Console.WriteLine("Enter the job number that needs to be closed")

        Try
            ' 1. read the job object
            Dim job As Job = readJob(Console.ReadLine())

            '2. Set the adminStatus to C
            job.adminStatus = "C"

            '3. Read the instance of UpdateObjectHttpBinding 
            Dim updateObjectHttpBinding As UpdateObjectHttpBinding = PaceClient.getUpdateObjectHttpBinding()

            '4. Update the job.
            updateObjectHttpBinding.updateJob(job)
            Console.WriteLine("Closed the Job Successfully")
        Catch ex As Exception
            Console.WriteLine(ex.Message)
        End Try

    End Sub

    Private Shared Function readJob(ByVal jobNumber As String) As Job
        Try
            Dim readObjectHttpBinding As ReadObjectHttpBinding = PaceClient.getReadObjectHttpBinding()
            Dim job As Job = New Job()
            job.job = jobNumber
            Return readObjectHttpBinding.readJob(job)
        Catch ex As SoapException
            Console.WriteLine(ex.Message)
            Return Nothing
        End Try
    End Function

End Class

File: docs/legacy(old PACE)/C#/PaceAPISample/sample/CloseJob.cs
using System;
using System.Web.Services.Protocols;

using efipaceservices;

namespace Pace_Web_Service_SDK
{
	/// <summary>
	/// Summary description for CloseJob.
	/// </summary>
	public class CloseJob
	{
		public static void Run()
		{
            ReadObjectPortType readObjectHttpBinding = PaceClient.getReadObjectHttpBinding();
            UpdateObjectPortType updateObjectHttpBinding = PaceClient.getUpdateObjectHttpBinding();

            Console.WriteLine("Enter the job number");

			Job job = new Job();
			job.job = Console.ReadLine(); // parameter
            try { job = readObjectHttpBinding.readJob(job); }
			catch (SoapException e)
			{
				Console.WriteLine("Job: " + job.job + " does not exist. Cannot close job that does not exist" + e.Message );
				return;
			}

            Console.WriteLine("Job exists, attempting to update");
            job.adminStatus = "C";
            updateObjectHttpBinding.updateJob(job);
            Console.WriteLine("Sent Job change request");

            job = readObjectHttpBinding.readJob(job);
            if (!"C".Equals(job.adminStatus))
            {
                throw new Exception("Job closing update failed");
            }
            else
            {
                Console.WriteLine("Read shows changed Job is closed");
            }
        }
	}
}
